### PR TITLE
upgrade kryo dependence to be compatible with OSGI.

### PR DESCRIPTION
**Pull request by hboumedane**
upgrade kryo dependence to be compatible with OSGI.
refactor the old code that uses the old kryo version.

### CODE REVIEW

**Comment by leleuj** [comment]
Thanks for the PR. I just released 1.8.0-RC1, but I think it can go in the final version.

+1

**Comment by hboumedane** [comment]
No problems, I will help when I can

**Comment by hboumedane** [comment]
can you merge it with the snapshot version ?

File: pac4j-cas/pom.xml [hunk 1]
```diff
@@ -88,11 +88,11 @@
 			<artifactId>logback-classic</artifactId>
 			<scope>test</scope>
 		</dependency>
-        <dependency>
-            <groupId>com.googlecode</groupId>
-            <artifactId>kryo</artifactId>
-            <scope>test</scope>
-        </dependency>
+		<dependency>
```
Comment (mmoayyed) [inline pac4j-cas/pom.xml]: Please fix the indentation. This is a problem with spaces vs tabs.

File: pac4j-core/pom.xml [hunk 2]
```diff
@@ -35,12 +35,11 @@
 			<groupId>javax.servlet</groupId>
 			<artifactId>javax.servlet-api</artifactId>
 		</dependency>
-        <dependency>
-            <groupId>com.googlecode</groupId>
-            <artifactId>kryo</artifactId>
-            <scope>provided</scope>
-        </dependency>
-
+		<dependency>
```
Comment (mmoayyed) [inline pac4j-core/pom.xml]: Please fix the indentation. This is a problem with spaces vs tabs.

**Comment by mmoayyed** [comment]
:+1:

**Comment by hboumedane** [comment]
how you want indentation ? space or tabs ?

**Comment by mmoayyed** [comment]
Always spaces, never tabs.

**Comment by mmoayyed** [comment]
The change will go into the latest SNAPSHOT once Jérôme merges it in. It looks good. Thank you.

**Comment by hboumedane** [comment]
indentation changed as asked :+1:

**Comment by leleuj** [comment]
Thanks for the PR

**Pull request merged**